mitosheet/src/jupyter/jupyterUtils.tsx

```diff
@@ -96,4 +96,34 @@ export const getArgs = (analysisToReplayName: string | undefined): Promise<strin
         }
         return resolve([]);
     })
+}
+
+export const writeToNotebookMetadata = (analysisName: string, key: string, value: string): void => {
+    if (isInJupyterLab()) {
+        window.commands?.execute('set-generated-code-cell-metadata', {
+            key: key,
+            value: value,
+            analysisName: analysisName
+        });
+    } else if (isInJupyterNotebook()) {
+        setGeneratedCodeCellMetadataInNotebook(analysisName, key, value);
+    } else {
+        console.error("Not detected as in Jupyter Notebook or JupyterLab")
+    }
+}
+
+export const getNotebookMetadata = (analysisName: string, key: string): Promise<void | string> => {
```

Comment:
Maybe we could be more specific that this isn't just getting the notebook metadata, but is getting an analysis from the notebook metatadata: `getAnalysisFromNotebookMetadata`